apps/src/makerlab/BoardController.js

```diff
@@ -0,0 +1,197 @@
+var five = require('johnny-five');
+var ChromeSerialPort = require('chrome-serialport');
+var PlaygroundIO = require('playground-io');
+require("babelify/polyfill"); // required for Promises in IE / Phantom
+
+/** @const {string} */
+var CHROME_APP_ID = 'himpmjbkjeenflliphlaaeggkkanoglo';
+
+var BoardController = module.exports = function () {
+  ChromeSerialPort.extensionId = CHROME_APP_ID;
+
+  /** @private {five.Board} */
+  this.board_ = null;
+
+  this.prewiredComponents = null;
+};
+
+BoardController.prototype.connectAndInitialize = function (codegen, interpreter) {
+  return this.ensureBoardConnected()
+      .then(this.installComponentsOnInterpreter.bind(this, codegen, interpreter))
+      .catch(function (reason) {
```

Comment:
conventionally, catch functions usually call this err or error (instead of reason)